util/src/main/java/tc/oc/pgm/util/attribute/AttributeInstance.java

```diff
@@ -0,0 +1,70 @@
+package tc.oc.pgm.util.attribute;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class AttributeInstance {
```

Comment:
Ideally, this is an interface. There is a `SportPaper` implementation, a `Spigot` one, and eventually we would have implementations in newer versions.

Reply:
We won't need a separate implementation for Spigot and SportPaper. I'll make an interface for this though.